Makefile

```diff
@@ -104,6 +97,20 @@ XCFLAGS += $(CFLAGS) $(INC_FLAGS) -D UT_VERSION=\"$(VERSION)\"
 VPATH += $(UT_DIR)
 VPATH += $(TOP_DIR)
 
+
+# Ensure the framework is built
+framework: $(eval SHELL:=/usr/bin/env bash)
+	@echo -e ${GREEN}"Ensure framework is present"${NC}
+	${UT_DIR}/build.sh
+	@echo -e ${GREEN}Completed${NC}
+	@if [ ! -d "${UT_DIR}/framework/ut-control/lib" ]; then \
```

Comment:
remove if the`if`, you should not be checking someone else scope.

need to define `UT_CONTROL` location.

`make -C ${UT_CONTROL}/ut-control`
`@cp ${UT_CONTROL}/lib/libut_control.*`